test/e2e/network_test.go

```diff
@@ -154,6 +155,51 @@ func testEastWestNetworking(t *testing.T) {
 	}
 }
 
+// testPodDNSResolution test the DNS resolution in a Windows pod
+func testPodDNSResolution(t *testing.T) {
+	// create context
+	testCtx, err := NewTestContext()
+	require.NoError(t, err)
+	// the following DNS resolution tests use curl.exe because nslookup tool is not present
+	// in the selected container image for the e2e tests. Ideally, we would use the native
+	// PowerShell cmdlet Resolve-DnsName, but it is not present either.
+	// TODO: Use a compatible container image for the e2e test suite that includes the
+	//  PowerShell cmdlet Resolve-DnsName.
+	t.Run("Valid URL", testCtx.testPodDNSResolutionWithValidUrl)
+	t.Run("Invalid URL", testCtx.testPodDNSResolutionWithInvalidUrl)
+}
+
+// testPodDNSResolutionWithValidUrl verifies that a curl query succeed using a valid URL
+// with resolvable DNS record
+func (tc *testContext) testPodDNSResolutionWithValidUrl(t *testing.T) {
+	// arrange
+	winJob, err := tc.createWindowsServerJob("win-dns-tester-valid-url",
+		// curl'ing with --head flag to fetch only the headers as a lightweight alternative
+		"curl.exe --head -k https://www.openshift.com:443",
+		nil)
+	require.NoError(t, err, "could not create Windows tester job")
+	defer tc.deleteJob(winJob.Name)
+	// act
+	err = tc.waitUntilJobSucceeds(winJob.Name)
+	// assert
+	assert.NoError(t, err, "Windows tester job failed")
+}
+
+// testPodDNSResolutionWithInvalidUrl verifies that a curl query fail using an invalid URL
+// with unresolvable DNS record
+func (tc *testContext) testPodDNSResolutionWithInvalidUrl(t *testing.T) {
+	// arrange
+	winJob, err := tc.createWindowsServerJob("win-dns-tester-invalid-url",
+		"curl.exe http://www.invalid-url-with-unresolvable-dns-record.test",
```

Comment:
Should we use `curl.exe -k` here as well to ensure the error is due to unresovlable address and not an SSL issue?

Reply:
No, no need to, bc 1) using `http` not `https` hence there is not SSL and 2) in the hypothetical case with `https` the query fails at the DNS resolution before hitting `https` protocol.

Reply:
I see, thanks for the explanation + info